src/types.ts

```diff
@@ -196,3 +197,20 @@ export interface Issuer {
   signer: Signer
   alg?: string
 }
+
+/**
+ * Represents the Verification Options that can be passed to the verifyPresentation Method.
+ * The verification will fail if given options are NOT satisfied.
+ */
+export interface VerifyPresentationOptions {
+  domain?: string
+  challenge?: string
+}
+
+/**
+ * Represents the Creation Options that can be passed to the createVerifiablePresentationJwt Method.
+ */
+export interface CreatePresentationOptions {
```

Comment:
This interface is identical to the `VerifyPresentationOptions`. Do you picture a future when they would be different?
Otherwise, I think they could be de-duplicated.

Also, during creation, these options can be supplied directly to the payload. Either as `aud` and `nonce` and used directly or as `challenge`/`domain` and transformed to the IANA equivalents.
The vp/vc create methods already do these kinds of transformations for other fields mentioned in [the data model](https://w3c.github.io/vc-data-model/#jwt-encoding)

Reply:
Yes exactly. CreateOptions and VerifyOptions are distinctly different concepts and they are just by coincidence identical at the moment. We could add verifyOptions that are exposed by verifyJWT (e.g. skewTime) to this object to make the difference more apparent. Therefore I'd be against joining the data model here.

So do I understand you correctly that in addition to the option I should also be able to handle
`payload.challenge` and `payload.domain` to be remapped and deleted automatically? Or should I do that in-place of the options?

Reply:
I guess we are also fine to allow library users to input `payload.nonce` and `payload.aud` for `challenge` and `domain` if they know the convention beforehand. However they must use `challenge` and `domain` options for verification always.

Reply:
The verifyPresentation options should definitely extend the lower level `verifyJWT ` options, if possible.

> So do I understand you correctly that in addition to the option I should also be able to handle
payload.challenge and payload.domain to be remapped and deleted automatically? Or should I do that in-place of the options?

I was suggesting not using create options at all, but to do the remapping from `payload.challenge` and `payload.domain`.
There doesn't seem to be consensus in the community around deleting the remapped originals. (I think vc.js expects the IANA fields to duplicate the W3C model, for example) For this reason, I'm not insisting on deletion, but suggesting it for the sake of agreement with the rest of the code that does perform remap & delete for the W3C VC -> IANA JWT fields.

Reply:
```
The verifyPresentation options should definitely extend the lower level verifyJWT options, if possible.
```
I think this would be a discussion outside of this PR.

Reply:
> I was suggesting not using create options at all, but to do the remapping from payload.challenge and payload.domain.

I agree and would replace the options with that logic.

Reply:
@mirceanis I actually decided against doing the remapping from the payload, since `domain` and `challenge` are no properties of a W3C VC. They ARE Properties of various VC-Proof sections e.g. `vc_payload.proof.challenge` and `vc_payload.proof.domain`, but I did not want to introduce that logic, because `transformPresentationInput` does not apply to any proof-specific remappings.